public/js/search.js

```diff
@@ -0,0 +1,66 @@
+$(function(config){
+  var algolia = algoliasearch(config.applicationId, config.apiKey);
+  var index = algolia.initIndex(config.indexName);
+
+  hideSearch();
+
+  $('.clear-search img, .search-results-container img').on('click', function() {
+    $('.sidebar-navigation--search input').val('');
+    hideSearch();
+  });
+
+  $('.sidebar-navigation--search input').on('input', function() {
+    var query = $(this).val();
+    onQueryChange(query);
+  });
+
+  function onQueryChange(query) {
+    if (query.length) {
+      index.search(query, {'hitsPerPage': '10', 'page': '0', 'typoTolerance': 'false'}, onResult);
+    } else {
+      hideSearch();
+    }
+  };
+
+  function onResult(err, data) {
+    if (err) { return console.error(err) }
+    displayResults(data);
+  };
+
+  function displayResults(data) {
+    showSearch();
+    $('.search-results').html('');
```

Comment:
You probably want to clear out the HTML before you call `showSearch`. Even if things are really fast or you're doing some clear out at another point in time, it's still good practice to make sure something is cleared out before you display it in the HTML. Prevents any scenario where you might have a quick flash of old results leftover in there before they refresh themselves.